.github/workflows/ci.yml

```diff
@@ -10,21 +10,40 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [ubuntu-latest, windows-latest, macos-latest]
-        python-version: ['2.7', '3.8', '3.9', '3.10']
+        os: [ubuntu-22.04, windows-latest, macos-latest]
+        python-version: ['2.7', '3.8', '3.9', '3.10', '3.11']
       fail-fast: false
     steps:
       - uses: actions/checkout@v3
+
+      - name: Set up Python for Ubuntu Python 2
+        if: ${{ matrix.os == 'ubuntu-22.04' && matrix.python-version == '2.7' }}
```

Comment:
It's a good idea, but I think this is getting a bit messy -- instead, could we please break this out into three different tests to avoid the "if" and "if not" cases? We could have three tests:

* Ubuntu on Python 2.7 (no matrix)
* Ubuntu on Python 3.x versions (matrix on Python version)
* macOS and Windows on all Python versions (matrix on OS and Python version)

It'll be slightly more lines/duplication, but simpler and clearer to maintain, I think.